optimum/utils/__init__.py

```diff
@@ -12,22 +12,62 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 import importlib.util
+import inspect
+from contextlib import contextmanager
+
+from packaging import version
 
 
 CONFIG_NAME = "config.json"
 
 _onnxruntime_available = importlib.util.find_spec("onnxruntime") is not None
```

Comment:
I would add all of these version checking utility functions to something like `utils/version_utils.py`, but this can be done on another PR.

Reply:
Ok I'll take care of that in another PR!